test/gtest/uct/ib/test_ib.cc

```diff
@@ -4,297 +4,257 @@
 * See file LICENSE for terms.
 */
 
-#include <common/test.h>
-#include <uct/uct_test.h>
-
-extern "C" {
-#include <poll.h>
-#include <uct/api/uct.h>
-#include <ucs/time/time.h>
-#include <uct/ib/base/ib_device.h>
-#include <uct/ib/base/ib_iface.h>
-#include <uct/ib/base/ib_md.h>
-}
+#include <uct/ib/test_ib.h>
 
 
-class test_uct_ib : public uct_test {
-public:
-    void initialize() {
-        uct_test::init();
+test_uct_ib::test_uct_ib() : uct_test() {
+    m_e1 = NULL;
+    m_e2 = NULL;
 
-        m_e1 = uct_test::create_entity(0);
-        m_entities.push_back(m_e1);
+    device_list = NULL;
+    num_devices = 0;
+    ibctx       = NULL;
+    ibdev       = NULL;
 
-        m_e2 = uct_test::create_entity(0);
-        m_entities.push_back(m_e2);
+    memset(&port_attr, 0, sizeof(port_attr));
 
-        m_e1->connect(0, *m_e2, 0);
-        m_e2->connect(0, *m_e1, 0);
+    initialized = false;
+
+    get_dev_name_and_port();
+}
 
-        test_uct_ib::ib_am_handler_counter = 0;
+test_uct_ib::~test_uct_ib() {
+    if (device_list != NULL) {
+        ibv_free_device_list(device_list);
+        device_list = NULL;
+        num_devices = 0;
     }
+}
 
-    typedef struct {
-        unsigned length;
-        /* data follows */
-    } recv_desc_t;
+/* don't call uct_test() here, because we don't
+ * want to init UCT for all tests */
+void test_uct_ib::init() {
+    ibctx = open_device(&ibdev);
```

Comment:
no need for open_device() function, just open the device here according to GetParam()